agent/engine/task_manager.go

```diff
@@ -148,6 +149,18 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO: Add new task resources provisioned state ?
+			err := mtask.SetupPlatformResources()
+			if err != nil {
```

Comment:
>  we don't have a way to obtain a `non-nil` condition here at this moment. 

That probably points to either the lack of abstractions or wrong abstractions where-in `SetupPlatformResources` can return what we expect it to return to test the failure scenarios.  

> I'd like to focus this PR on `resources/cgroup`. 

If you're not planning to add tests to this code path in this PR, please remove that and add it back with tests in the PR where you plan to add tests.